src/acl.c

```diff
@@ -1419,16 +1665,13 @@ void ACLKillPubsubClientsIfNeeded(user *u, list *upcoming) {
  *
  * If the user can execute the command ACL_OK is returned, otherwise 
  * ACL_DENIED_CHANNEL. */
-int ACLCheckPubsubPerm(const user *u, robj **argv, int idx, int count, int literal, int *idxptr) {
-    /* If there is no associated user, the connection can run anything. */
-    if (u == NULL) return ACL_OK;
-
-    /* Check if the user can access the channels mentioned in the command's
-     * arguments. */
-    if (!(u->flags & USER_FLAG_ALLCHANNELS)) {
-        for (int j = idx; j < idx+count; j++) {
-            if (ACLCheckPubsubChannelPerm(argv[j]->ptr,u->channels,literal)
-                != ACL_OK) {
+int ACLSelectorCheckPubsub(aclSelector *s, robj **argv, int idx, int count, int literal, int *idxptr) {    
+    /* Nothing to kill when one of the selectors has all channel permissions. */
+    for (int j = idx; j < idx+count; j++) {
+        if (!(s->flags & SELECTOR_FLAG_ALLCHANNELS)) {
+            if (ACLCheckPubsubChannelPerm(argv[j]->ptr,s->channels,literal)
+                != ACL_OK) 
+            {
```

Comment:
i'd rather avoid these two line breaks (since we're changing these lines anyway due to indentation)